### PR TITLE
Add Prettier formatting

### PR DESCRIPTION
**Pull request by tsa96**
Adds a prettier config file and sets up Husky and lint-staged to run after each commit.

Adds Prettier scripts to package.json including a check for if we ever want to integrate it with Github Actions (probably overkill, up to you Goc).

Reformats all our JS, XML, SCSS and Markdown with Prettier. 

God it feels good to be back to Github PRs.

### CODE REVIEW

**Comment by Gocnak** [comment]
Let's add the github action to check lint output in this PR.

**Comment by tsa96** [comment]
Added the workflow for the action, and tested it using [Act](https://github.com/nektos/act)

File: .github/workflows/format.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+name: Formatting
+
+on:
+  push:
+    branches: [ master ]
```
Comment (Gocnak) [inline .github/workflows/format.yml:5]: Doesn't need to happen on push to master, only on PRs to master, as we're doing trunk-based development.

File: .github/workflows/format.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Use Node.js 18
+      uses: actions/setup-node@v3
+      with:
+        node-version: '18'
+        cache: 'npm'
+    - run: npm ci
```
Comment (Gocnak) [inline .github/workflows/format.yml:19]: What does `npm ci` do?

**Review by Gocnak — CHANGES_REQUESTED** [review]

File: .github/workflows/format.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Use Node.js 18
```
Comment (Gocnak) [inline .github/workflows/format.yml:14]: Nit
```suggestion
    - name: Setup Node.js 18

```

File: .github/workflows/format.yml (hunk 2 above)
Comment (tsa96) [reply .github/workflows/format.yml]: Was getting a bit confused over if I should be running `npm ci` or `npm install`. Been reading up on it and I think the fastest setup for us is first caching the `node_modules/` then running `npm install`, which grabs the dependencies if they're not cached.

File: .github/workflows/format.yml (hunk 3 above)
Comment (tsa96) [reply .github/workflows/format.yml]: Done

File: .github/workflows/format.yml (hunk 1 above)
Comment (tsa96) [reply .github/workflows/format.yml]: Done

**Review by Gocnak — APPROVED** [review]

**Pull request merged**